flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaInternalProducerITCase.java

```diff
@@ -148,62 +153,78 @@ public void testResumeTransaction() throws Exception {
 
-    @Test(timeout = 30000L, expected = IllegalStateException.class)
-    public void testPartitionsForAfterClosed() {
+    @Test
+    @Timeout(30L)
+    void testPartitionsForAfterClosed() {
         FlinkKafkaInternalProducer<String, String> kafkaProducer =
                 new FlinkKafkaInternalProducer<>(extraProperties);
         kafkaProducer.close(Duration.ofSeconds(5));
-        kafkaProducer.partitionsFor("Topic");
+        assertThatThrownBy(() -> kafkaProducer.partitionsFor("Topic"))
+                .isInstanceOf(IllegalStateException.class);
     }
 
-    @Test(timeout = 30000L, expected = IllegalStateException.class)
-    public void testInitTransactionsAfterClosed() {
+    @Test
+    @Timeout(30L)
+    void testInitTransactionsAfterClosed() {
         FlinkKafkaInternalProducer<String, String> kafkaProducer =
                 new FlinkKafkaInternalProducer<>(extraProperties);
         kafkaProducer.close(Duration.ofSeconds(5));
-        kafkaProducer.initTransactions();
+        assertThatThrownBy(kafkaProducer::initTransactions)
+                .isInstanceOf(IllegalStateException.class);
     }
 
-    @Test(timeout = 30000L, expected = IllegalStateException.class)
-    public void testBeginTransactionAfterClosed() {
+    @Test
+    @Timeout(30L)
+    void testBeginTransactionAfterClosed() {
         FlinkKafkaInternalProducer<String, String> kafkaProducer =
                 new FlinkKafkaInternalProducer<>(extraProperties);
         kafkaProducer.initTransactions();
         kafkaProducer.close(Duration.ofSeconds(5));
-        kafkaProducer.beginTransaction();
+        assertThatThrownBy(kafkaProducer::beginTransaction)
+                .isInstanceOf(IllegalStateException.class);
     }
 
-    @Test(timeout = 30000L, expected = IllegalStateException.class)
-    public void testCommitTransactionAfterClosed() {
+    @Test
+    @Timeout(30L)
+    void testCommitTransactionAfterClosed() {
         String topicName = "testCommitTransactionAfterClosed";
         FlinkKafkaInternalProducer<String, String> kafkaProducer = getClosedProducer(topicName);
-        kafkaProducer.commitTransaction();
+        assertThatThrownBy(kafkaProducer::commitTransaction)
+                .isInstanceOf(IllegalStateException.class);
     }
 
-    @Test(timeout = 30000L, expected = IllegalStateException.class)
-    public void testResumeTransactionAfterClosed() {
+    @Test
+    @Timeout(30L)
+    void testResumeTransactionAfterClosed() {
         String topicName = "testAbortTransactionAfterClosed";
         FlinkKafkaInternalProducer<String, String> kafkaProducer = getClosedProducer(topicName);
-        kafkaProducer.resumeTransaction(0L, (short) 1);
+        assertThatThrownBy(() -> kafkaProducer.resumeTransaction(0L, (short) 1))
+                .isInstanceOf(IllegalStateException.class);
     }
 
-    @Test(timeout = 30000L, expected = IllegalStateException.class)
-    public void testAbortTransactionAfterClosed() {
+    @Test
+    @Timeout(30L)
```

Comment:
time unit should be specified

Reply:
By default it takes seconds, do we still need to specify?